source/user/mp.rst

```diff
@@ -5,18 +5,134 @@ Second-Order Møller–Plesset Perturbation Theory
 
 *Modules*: :mod:`mp`, :mod:`pbc.mp`
 
+The interfaces to coupled cluster and MP2 calculations look very similar
+in PySCF.  Therefore, note the similarities of this section to the
+:ref:`theory_cc` documentation.
+
+
+Introduction
+============
+
+Second-Order Møller–Plesset Perturbation Theory (MP2) :cite:`Moller1934`
+is a post-Hartree--Fock method.
+
+A simple example (see :source:`examples/mp/00-simple_mp2.py`) of running
+an MP2 calculation is
+
+.. literalinclude:: ../../examples/mp/00-simple_mp2.py
+
+which outputs
+
+.. code::
+
+  converged SCF energy = -99.9873974403487
+  E(MP2) = -100.198764900659  E_corr = -0.211367460310054
+
+the Hartree--Fock energy, the MP2 energy and their difference, the
+MP2 correlation energy.
+
+.. note::
+
+  The last line in the code example above could have been replaced by
+
+  .. code ::
+    
+    pyscf.mp.MP2(mf).kernel()
+  
+  for the same result.
+
+
+Spin symmetry
+=============
+
+The MP2 module in PySCF supports a number of broken spin symmetry reference
+wavefunctions.  In particular, MP2 can be performed with a spin-restricted,
+spin-unrestricted, and general (spin-mixed) Hartree-Fock solution, leading
+to the RMP2, UMP2, and GMP2 methods.
+
+The module-level ``mp.MP2(mf)`` constructor can infer the correct method based
+on the level of symmetry-breaking in the mean-field argument.  For more explicit
+control or inspection, the respective classes and functions can be found in
+``mp2.py`` (restricted), ``ump2.py`` (unrestricted), and ``gmp2.py``
+(general).
+
+For example, a spin-unrestricted calculation on triplet oxygen can be performed
+as follows::
+
+    from pyscf import gto, scf, mp
+    mol = gto.M(
+        atom = 'O 0 0 0; O 0 0 1.2',  # in Angstrom
+        basis = 'ccpvdz',
+        spin = 2
+    )
+    mf = scf.HF(mol).run() # this is UHF
+    mymp = mp.MP2(mf).run() # this is UMP2
+    print('UMP2 total energy = ', mymp.e_tot)
+
+
+Properties
+==========
+
+A number of properties are available at the MP2 level.
+
+Unrelaxed 1- and 2-electron reduced density matrices can be calculated. 
+They are returned in the MO basis::
+
+    dm1 = mymp.make_rdm1()
+    dm2 = mymp.make_rdm2()
+
+Gradients (see e.g. :cite:`Pople1979`, :cite:`Handy1985` for MP2 gradients)
+can be calculated::
+
+    from pyscf import grad
+    mygrad = mymp.Gradients()
+    grad = mygrad.kernel()
+
+
+Frozen orbitals
+===============
+
+By default, MP2 calculations are performed in PySCF without any frozen
+orbitals, including core orbitals. To freeze the lowest-energy core
```

Comment:
By default, MP2 calculations in PySCF include all electrons; that is, also core orbitals are correlated.